File: src/ObjectManager/Object.Tes/FilePacks/Records/045-EYES.Eyes.cs
﻿using OA.Core;
using System;

namespace OA.Tes.FilePacks.Records
{
    public class EYESRecord : Record
    {
        public override string ToString() => $"EYES: {EDID.Value}";
        public STRVField EDID;
        public STRVField FULL;
        public FILEField ICON;
        public BYTEField DATA; // Playable

        public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, uint dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "FULL": FULL = new STRVField(r, dataSize); return true;
                case "ICON": ICON = new FILEField(r, dataSize); return true;
                case "DATA": DATA = new BYTEField(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/045-EYES.Eyes.swift
﻿//
//  EYESRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class EYESRecord: Record {
    public var description: String { return "EYES: \(EDID)" }
    public var EDID: STRVField // Editor ID
    public var FULL: STRVField
    public var ICON: FILEField
    public var DATA: BYTEField // Playable

    init() {
    }
    
    override func createField(r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = STRVField(r, dataSize)
        case "FULL": FULL = STRVField(r, dataSize)
        case "ICON": ICON = FILEField(r, dataSize)
        case "DATA": DATA = BYTEField(r, dataSize)
        default: return false
        }
        return true
    }
}
